### PR TITLE
Add experimental implementation, using the AST from the Skylark interpreter

### PR DESCRIPTION
**Pull request by laurentlb**

### CODE REVIEW

**Comment by vladmos** [comment]
Seems good, but maybe add a simple smoke test that runs the new binary and checks that at least a simple file is formatted correctly? Like the one in #237.

**Comment by laurentlb** [comment]
Added simple test

File: buildifier/buildifier2.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/bazelbuild/buildtools/build"
+	"github.com/google/skylark/syntax"
+)
+
+func main() {
+	flag.Parse()
+
+	switch len(flag.Args()) {
+	case 0:
+		fmt.Println("Argument missing")
```
Comment (pmbethe09) [inline buildifier/buildifier2.go]: log.Info?

File: buildifier/convert_ast.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
+// This file contains functions to convert from one AST to the other.
+// Input: AST from github.com/google/skylark/syntax
+// Output: AST from github.com/bazelbuild/buildtools/build
+
+package main
```
Comment (pmbethe09) [inline buildifier/convert_ast.go]: I think this file belongs in its own
package astconvert 
with some of these functions public

File: buildifier/convert_ast.go [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+	}
+	switch e := e.(type) {
+	case *syntax.Literal:
+		switch e.Token {
+		case syntax.INT:
+			return &build.LiteralExpr{
+				Token:    strconv.FormatInt(e.Value.(int64), 10),
+				Comments: convComments(e.Comments())}
+		case syntax.FLOAT:
+			panic("float")
```
Comment (pmbethe09) [inline buildifier/convert_ast.go]: prefer log.Fatal

**Pull request merged**